test/unit/components/useIntl.js

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import {mount} from 'enzyme';
+import {IntlProvider} from '../../../src/react-intl'
+import useIntl from '../../../src/components/useIntl';
+
+const FunctionComponent = ({ spy }) => {
+  const intl = useIntl();
+  spy(intl);
+  return null;
+};
+
+describe('useIntl() hook', () => {
+  it('throws when <IntlProvider> is missing from ancestry', () => {
+    const consoleError = jest.spyOn(console, 'error').mockImplementation(() => {}); // surpress console error from JSDom
+    expect(() => mount(<FunctionComponent />)).toThrow(
+      '[React Intl] Could not find required `intl` object. <IntlProvider> needs to exist in the component ancestry.'
+    );
+    consoleError.mockRestore();
+  });
+
+  it('spy on the returned value from hook', () => {
```

Comment:
We try to make the test names describe what is being tested (and ideally use the it `it` in the sentence), e.g.:
```js
it('hooks onto the intl context')
```
This makes the test status overview a lot easier to read :smile:

Reply:
Nice test naming suggestion! It made a lot sense now 🚀!